server/providers/CustomProviderAdapter.js

```diff
@@ -0,0 +1,70 @@
+const Database = require('../Database')
+const axios = require("axios");
+const Logger = require("../Logger");
+
+class CustomProviderAdapter {
+    constructor() {
+    }
+
+    async search(title, author, providerSlug) {
+        const providerId = providerSlug.split("custom-")[1]
+        const provider = await Database.customMetadataProviderModel.findByPk(providerId);
+
+        if (!provider) {
+            throw new Error("Custom provider not found for the given id");
+        }
+
+        const matches = await axios.get(`${provider.url}/search?query=${encodeURIComponent(title)}${!!author ? `&author=${encodeURIComponent(author)}` : ""}`, {
+            headers: {
+                "Authorization": provider.apiKey,
```

Comment:
If a user is using bearer token authorization or basic authorization would they be instructed to enter that in the `apiKey` input? Let me know if that doesn't make sense

Reply:
Basic auth isn't supported in the current model, as it would require 2 pieces of credentials, but bearer, is essentially just Authorization header + a Bearer prefix and an api key so they could make that work

Reply:
Basic auth would be supported in this model as far as I understand it because the user is required to enter the full authorization header value.
`Basic <credentials>` or `Bearer <api key>`.

My question was about if we are requiring the user to write in the `Bearer` part of the authorization header. If we are then `apiKey` might not be an accurate name for the variable.